Libraries/LibWeb/Compression/CompressionStream.cpp

```diff
@@ -113,21 +114,22 @@ WebIDL::ExceptionOr<void> CompressionStream::compress_and_enqueue_chunk(JS::Valu
         return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, "Chunk is not a BufferSource type"sv };
 
     // 2. Let buffer be the result of compressing chunk with cs's format and context.
-    auto buffer = [&]() -> ErrorOr<ByteBuffer> {
+    auto maybe_buffer = [&]() -> ErrorOr<ByteBuffer> {
         if (auto buffer = WebIDL::underlying_buffer_source(chunk.as_object()))
             return compress(buffer->buffer(), Finish::No);
         return ByteBuffer {};
     }();
+    if (maybe_buffer.is_error())
+        return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, MUST(String::formatted("Unable to compress chunk: {}", maybe_buffer.error())) };
 
-    if (buffer.is_error())
-        return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, MUST(String::formatted("Unable to compress chunk: {}", buffer.error())) };
+    auto buffer = maybe_buffer.release_value();
 
     // 3. If buffer is empty, return.
-    if (buffer.value().is_empty())
+    if (buffer.is_empty())
         return {};
 
     // 4. Split buffer into one or more non-empty pieces and convert them into Uint8Arrays.
-    auto array_buffer = JS::ArrayBuffer::create(realm, buffer.release_value());
+    auto array_buffer = JS::ArrayBuffer::create(realm, buffer);
```

Comment:
I haven't looked through all of this, but here you are now creating an unnecessary copy of the buffer. You should `move` it in.